src/sherpa_ai/actions/google_search.py

```diff
@@ -31,10 +33,18 @@ def __init__(
         self.description = description
         self.llm = llm
         self.n = n
+        self.config = config
 
         self.search_tool = SearchTool()
 
+    def config_gsite_query(self, query) -> str:
+        # check if the gsite is none
+        if self.config.gsite:
+            query = query + " site:" + self.config.gsite
+        return query
+
     def execute(self, query) -> str:
+        query = self.config_gsite_query(query)
```

Comment:
This code adds `site:` to the query by calling `config_gsite_query`.
Then SearchTool._run adds `site:` to the query by calling `query_with_gsqite`.  
(Assuming they are both using the same config.)
We should just add `site` to the query once, either here in `GoogleSearch` or in `SearchTool`.

If gsite is intended to be specific to google search then do the work here in `GoogleSearch`. 

Either way I suggest calling the function `query_with_gsite` or something like that.

Reply:
@YujingYang666777

Reply:
OK, I had a look at the implementation. It seems that this feature is meant to be added to the SearchTool, so the one in the GoogleSearch action is removed. And I moved the test accordingly.